### PR TITLE
fix(zc): triggered autowarp combos use sidewarps, screenflags for direct autowarps and sidewarps work when respective combo is triggered.

### PR DESCRIPTION
**Pull request by Jambu2024**

### CODE REVIEW

File: src/zc/combos.cpp [hunk 1]
```diff
@@ -968,12 +968,23 @@ bool trigger_warp(newcombo const& cmb)
 	mapscr* wscr2=&TheMaps[ws];
 	auto wx = wscr2->warpreturnx[wrindex];
 	auto wy = wscr2->warpreturny[wrindex];
```
Comment (connorjclark) [inline src/zc/combos.cpp]:
```suggestion
	int wx = wscr2->warpreturnx[wrindex];
	int wy = wscr2->warpreturny[wrindex];
```

File: src/zc/hero.cpp [hunk 2]
```diff
@@ -24487,6 +24487,8 @@ void HeroClass::checkspecial2(int32_t *ls)
 		bool hasStep[4] = {false};
 		for(auto p = 0; p < 4; ++p)
 		{
+			if ((z > 0 || fakez > 0) && !(tmpscr->flags2 & fAIRCOMBOS))
+				break;
```
Comment (connorjclark) [inline src/zc/hero.cpp]: is this related to warps?

File: src/zc/combos.cpp [hunk 3]
```diff
@@ -968,12 +968,23 @@ bool trigger_warp(newcombo const& cmb)
...
+	if (stype == cAWARPA) //these use sidewarps
+	{
+		wtype = wscr->sidewarptype[index];
+		if (wtype == wtNOWARP)
+			return false;
+		wsfx = cmb.attribytes[0];
+		tdm = wscr->sidewarpdmap[index];
+		tscr = wscr->sidewarpscr[index];
+		ws = (DMaps[tdm].map * MAPSCRS + wscr->sidewarpscr[index] + DMaps[tdm].xoff);
+		wscr2 = &TheMaps[ws];
```
Comment (connorjclark) [inline src/zc/combos.cpp:981]: wscr2 isn't used after this line, did you expect it to be somehow?

File: src/zc/hero.cpp (hunk 2 above)
Comment (Jambu2024) [reply src/zc/hero.cpp]: yes it causes the combo to effect link when in air

File: src/zc/combos.cpp (hunk 3 above)
Comment (Jambu2024) [reply src/zc/combos.cpp:981]: fixed

File: src/zc/combos.cpp (hunk 1 above)
Comment (Jambu2024) [reply src/zc/combos.cpp]: these are not ints

Comment (connorjclark) [reply src/zc/combos.cpp]: but `wx = Hero.getX().getInt();` is - so i'm worried about under/overflow, which could happen now and may happen in z3.

File: src/zc/combos.cpp [hunk 4]
```diff
@@ -968,12 +968,25 @@ bool trigger_warp(newcombo const& cmb)
...
+		wx = wscr2->warpreturnx[wrindex];
```
Comment (EmilyV99) [inline src/zc/combos.cpp:982]: This whole if needs to be ABOVE the check for direct warp, because you're just setting wx/wy here (overwriting the direct warp above)

File: src/zc/combos.cpp (hunk 1 above)
Comment (EmilyV99) [reply src/zc/combos.cpp]: Agree with Connor here.

File: src/zc/hero.cpp (hunk 2 above)
Comment (EmilyV99) [reply src/zc/hero.cpp]: This is step combos, not warps? It is a correct bugfix I believe, but not related to warps.

This would appear to be a correct bugfix though...for [this bug](https://discord.com/channels/876899628556091432/1275994756266197053) which was already marked "closed"?

Comment (Jambu2024) [reply src/zc/hero.cpp]: triggers tab to be more specific.

**Comment by Jambu2024** [comment]
changelog summary of everything in first post.
